src/main/java/org/traccar/protocol/HuabaoProtocolDecoder.java

```diff
@@ -772,6 +775,11 @@ private Position decodeLocation(DeviceSession deviceSession, ByteBuf buf) {
                         }
                     }
                     break;
+                case 0xF7:
+                    position.set(Position.KEY_BATTERY, buf.readUnsignedInt() * 0.001);
+                    position.set(Position.KEY_CHARGE, buf.readUnsignedByte() >= 2);
```

Comment:
This doesn't look correct. The documentation says that only 2 and 3 indicate charing.